third_party/silabs/efr32_sdk.gni

```diff
@@ -480,6 +481,10 @@ template("efr32_sdk") {
           # TODO: Remove this flag, once the communication is fixed
           "WIFI_DEBUG_ENABLED=1",
         ]
+
+        # This is kept due to the warning on the LWIP when on demand timer is added
+        # TODO: remove this flag once the warning is fixed in SiSDK MATTER-3946
+        cflags += [ "-Wno-error" ]
```

Comment:
Is there not a more targetted flag for the error you need to ignore temporarily? No-error seems way to general and will hide other major issues

Reply:
Yeah, modified to cflags_c and added the required warning of implicit function declaration, since it cannot be used in the cpp